qutip/nonmarkov/bofin_solvers.py

```diff
@@ -455,10 +455,16 @@ def __init__(
             for k in range(self._n_exponents)
         ]
 
+        self.progress_bar = progress_bar
         if progress_bar is None:
             self.progress_bar = BaseProgressBar()
         if progress_bar is True:
             self.progress_bar = TextProgressBar()
+        if not isinstance(self.progress_bar, BaseProgressBar):
```

Comment:
Thanks for sorting this out. I would recommend we change it a little more to be:
```
if progress_bar is None:
    ...
elif progress_bar is True:
    ...
elif isinstance(self.progress_bar, BaseProgressBar):
    ...
else:
    raise TypeError(...)
```
just to make it easy to see that the logic is correct.

Reply:
Thanks for the feedback :) I changed the if-statement to this cleaner version.